wapitiCore/attack/attack.py

```diff
@@ -246,34 +246,47 @@ def __init__(
+            payload_type=finding_class.type(),
             module=self.name,
-            category=category,
+            category=finding_class.name(),
             level=level,
             request=request,
             parameter=parameter,
             info=info,
-            wstg=wstg,
+            wstg=finding_class.wstg_code(),
             response=response
         )
 
-    add_vuln = partialmethod(add_payload, payload_type=VULN)
-    add_vuln_critical = partialmethod(add_payload, payload_type=VULN, level=CRITICAL_LEVEL)
-    add_vuln_high = partialmethod(add_payload, payload_type=VULN, level=HIGH_LEVEL)
-    add_vuln_medium = partialmethod(add_payload, payload_type=VULN, level=MEDIUM_LEVEL)
-    add_vuln_low = partialmethod(add_payload, payload_type=VULN, level=LOW_LEVEL)
-    add_vuln_info = partialmethod(add_payload, payload_type=VULN, level=INFO_LEVEL)
+    # Define explicit wrapper functions for each severity level
+    async def add_info(self, finding_class: Type[FindingBase], request_id: int = -1,
+                       request: Optional[Request] = None, parameter: str = "",
+                       info: str = "", response: Optional[Response] = None):
+        await self.add_payload(finding_class, request_id, INFO_LEVEL, request, parameter, info, response)
 
-    add_anom_high = partialmethod(add_payload, payload_type=ANOM, level=HIGH_LEVEL)
-    add_anom_medium = partialmethod(add_payload, payload_type=ANOM, level=MEDIUM_LEVEL)
+    async def add_low(self, finding_class: Type[FindingBase], request_id: int = -1,
+                      request: Optional[Request] = None, parameter: str = "",
+                      info: str = "", response: Optional[Response] = None):
+        await self.add_payload(finding_class, request_id, LOW_LEVEL, request, parameter, info, response)
 
-    add_addition = partialmethod(add_payload, payload_type=ADDITION, level=INFO_LEVEL)
+    async def add_medium(self, finding_class: Type[FindingBase], request_id: int = -1,
+                         request: Optional[Request] = None, parameter: str = "",
+                         info: str = "", response: Optional[Response] = None):
+        await self.add_payload(finding_class, request_id, MEDIUM_LEVEL, request, parameter, info, response)
+
+    async def add_high(self, finding_class: Type[FindingBase], request_id: int = -1,
+                       request: Optional[Request] = None, parameter: str = "",
+                       info: str = "", response: Optional[Response] = None):
+        await self.add_payload(finding_class, request_id, HIGH_LEVEL, request, parameter, info, response)
+
+    async def add_critical(self, finding_class: Type[FindingBase], request_id: int = -1,
+                           request: Optional[Request] = None, parameter: str = "",
+                           info: str = "", response: Optional[Response] = None):
+        await self.add_payload(finding_class, request_id, CRITICAL_LEVEL, request, parameter, info, response)
```

Comment:
The point of the MR is precisely to prevent bad data to be pushed to the persister. I don't see any benefit in keeping those functions as the type is now held by the finding. A XSS is a vulnerability and will always be. An internal error is an anomaly and should never change. The purpose is to prevent contributors to use a different type or invalid wstg. ONly the severity level can be chosen.